### ISSUE TITLE
"Failed to load json data" error

### ISSUE THREAD

**Issue by ManOnDaMoon** [Daniel-KM/Omeka-S-module-Timeline#12]
Hello,

I've tried to use the update with latest Omeka S release and I get an error when the Simile timeline loads:
"Failed to load json data from /timeline/10/events.json
Internal Server Error"

When browsing to the timeline block URL, I get the following stacktrace:
```
Omeka S a rencontré une erreur
Doctrine\ORM\Query\QueryException
[Syntax Error] line 0, col 7: Error: Expected IdentificationVariable | ScalarExpression | AggregateExpression | FunctionDeclaration | PartialObjectExpression | "(" Subselect ")" | CaseExpression, got 'Omeka\Entity\SitePageBlock'

Détails :

Doctrine\ORM\Query\QueryException: SELECT Omeka\Entity\SitePageBlock FROM Omeka\Entity\SitePageBlock Omeka\Entity\SitePageBlock WHERE Omeka\Entity\SitePageBlock.id = :id in /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/QueryException.php:43
Stack trace:
#0 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(456): Doctrine\ORM\Query\QueryException::dqlError('SELECT Omeka\\En...')
#1 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2253): Doctrine\ORM\Query\Parser->syntaxError('IdentificationV...', Array)
#2 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1182): Doctrine\ORM\Query\Parser->SelectExpression()
#3 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(878): Doctrine\ORM\Query\Parser->SelectClause()
#4 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(847): Doctrine\ORM\Query\Parser->SelectStatement()
#5 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(260): Doctrine\ORM\Query\Parser->QueryLanguage()
#6 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(359): Doctrine\ORM\Query\Parser->getAST()
#7 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(283): Doctrine\ORM\Query\Parser->parse()
#8 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(295): Doctrine\ORM\Query->_parse()
#9 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(967): Doctrine\ORM\Query->_doExecute()
#10 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(922): Doctrine\ORM\AbstractQuery->executeIgnoreQueryCache(NULL, NULL)
#11 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(765): Doctrine\ORM\AbstractQuery->execute(NULL, NULL)
#12 /Users/laurent/Documents/Collection/omeka-s/modules/Timeline/src/Controller/TimelineController.php(61): Doctrine\ORM\AbstractQuery->getOneOrNullResult()
#13 /Users/laurent/Documents/Collection/omeka-s/modules/Timeline/src/Controller/TimelineController.php(25): Timeline\Controller\TimelineController->getBlock(10)
#14 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(78): Timeline\Controller\TimelineController->eventsAction()
#15 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#16 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#17 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(104): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#18 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/DispatchListener.php(138): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Http\PhpEnvironment\Request), Object(Zend\Http\PhpEnvironment\Response))
#19 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#20 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#21 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/Application.php(332): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#22 /Users/laurent/Documents/Collection/omeka-s/index.php(21): Zend\Mvc\Application->run()
#23 {main}

Next Doctrine\ORM\Query\QueryException: [Syntax Error] line 0, col 7: Error: Expected IdentificationVariable | ScalarExpression | AggregateExpression | FunctionDeclaration | PartialObjectExpression | "(" Subselect ")" | CaseExpression, got 'Omeka\Entity\SitePageBlock' in /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/QueryException.php:54
Stack trace:
#0 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(456): Doctrine\ORM\Query\QueryException::syntaxError('line 0, col 7: ...', Object(Doctrine\ORM\Query\QueryException))
#1 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(2253): Doctrine\ORM\Query\Parser->syntaxError('IdentificationV...', Array)
#2 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(1182): Doctrine\ORM\Query\Parser->SelectExpression()
#3 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(878): Doctrine\ORM\Query\Parser->SelectClause()
#4 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(847): Doctrine\ORM\Query\Parser->SelectStatement()
#5 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(260): Doctrine\ORM\Query\Parser->QueryLanguage()
#6 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Parser.php(359): Doctrine\ORM\Query\Parser->getAST()
#7 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(283): Doctrine\ORM\Query\Parser->parse()
#8 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/Query.php(295): Doctrine\ORM\Query->_parse()
#9 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(967): Doctrine\ORM\Query->_doExecute()
#10 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(922): Doctrine\ORM\AbstractQuery->executeIgnoreQueryCache(NULL, NULL)
#11 /Users/laurent/Documents/Collection/omeka-s/vendor/doctrine/orm/lib/Doctrine/ORM/AbstractQuery.php(765): Doctrine\ORM\AbstractQuery->execute(NULL, NULL)
#12 /Users/laurent/Documents/Collection/omeka-s/modules/Timeline/src/Controller/TimelineController.php(61): Doctrine\ORM\AbstractQuery->getOneOrNullResult()
#13 /Users/laurent/Documents/Collection/omeka-s/modules/Timeline/src/Controller/TimelineController.php(25): Timeline\Controller\TimelineController->getBlock(10)
#14 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/Controller/AbstractActionController.php(78): Timeline\Controller\TimelineController->eventsAction()
#15 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\Controller\AbstractActionController->onDispatch(Object(Zend\Mvc\MvcEvent))
#16 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#17 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/Controller/AbstractController.php(104): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#18 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/DispatchListener.php(138): Zend\Mvc\Controller\AbstractController->dispatch(Object(Zend\Http\PhpEnvironment\Request), Object(Zend\Http\PhpEnvironment\Response))
#19 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(322): Zend\Mvc\DispatchListener->onDispatch(Object(Zend\Mvc\MvcEvent))
#20 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-eventmanager/src/EventManager.php(179): Zend\EventManager\EventManager->triggerListeners(Object(Zend\Mvc\MvcEvent), Object(Closure))
#21 /Users/laurent/Documents/Collection/omeka-s/vendor/zendframework/zend-mvc/src/Application.php(332): Zend\EventManager\EventManager->triggerEventUntil(Object(Closure), Object(Zend\Mvc\MvcEvent))
#22 /Users/laurent/Documents/Collection/omeka-s/index.php(21): Zend\Mvc\Application->run()
#23 {main}
```

Same error is raised with Knightlab.

I tested with a previously working block configuration, and also with a default block config with the same result.

**Comment by Daniel-KM** [comment]
The compatibility with Omeka 2 is fixed.

**Comment by ManOnDaMoon** [comment]
Sorry @Daniel-KM, I'm afraid I still have the issue even using the latest release. Same error, with default block configuration. All other modules disabled.

Please tell if I can provide any information, or if there is anything that can be tested against my installation.

**Comment by Daniel-KM** [comment]
Can you retry again with release 3.4.10 (7f575e0) ? Thanks.

**Comment by ManOnDaMoon** [comment]
> Can you retry again with release 3.4.10 ([7f575e0](https://github.com/Daniel-KM/Omeka-S-module-Timeline/commit/7f575e0bbaa77830662b8959b1d2a42149c8b22b)) ? Thanks.

I tried again, the error is now gone, thanks. Unfortunately I still do not get any data back: the displayed timeline is empty, the `events` variable returned in the json query is empty.

Whenever the TimelineData service is called, my search query parameters are correctly passed, but the API search call returns no result - although the query parameters used to work. I tried a new empty query or a new query with only `item_set_id[]=7` with no luck.
